medusa/providers/torrent/html/shanaproject.py

```diff
@@ -0,0 +1,186 @@
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=20)
+
+    def search(self, search_strings, age=0, ep_obj=None):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if self.show and not self.show.is_anime:
+            return results
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                    search_params = {
+                        'sort': 'date',
+                        'dir': 'Descending',
+                        'title': '{0}'.format(search_string)
+                    }
+
+                page = 1
+                while page and page <= self.max_pages:
+                    response = self.get_url(
+                        self.urls['search'].format(page=page), params=search_params, returns='response'
+                    )
+                    if not response or not response.text:
+                        log.debug('No data returned from provider')
+                        page = None
+                        continue
+
+                    page = page + 1 if 'list_next >>' in response.text else None
+
+                    results += self.parse(response.text, mode)
+
+        return results
+
+    def parse(self, data, mode):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+
+            torrent_rows = html('div', {'class': 'release_block'})
+            if len(torrent_rows) < 2:
+                return
+
+            for row in torrent_rows[1:]:
+
+                try:
+                    first_cell = row.find('div', {'class': 'release_row_first'})
+                    cells = row('div', {'class': 'release_row'})
+
+                    title = cells[1].find('div', {'class': 'release_text_contents'}).get_text().strip()
+                    download_url = first_cell('a')[-1].get('href')
+
+                    if not all([title, download_url]):
+                        continue
+
+                    download_url = urljoin(self.url, download_url)
+
+                    # Provider does not support seeders or leechers.
+                    seeders = -1
+                    leechers = -1
+
+                    torrent_size = first_cell.find('div', {'class': 'release_size'}).get_text()
+                    size_regex = re.compile('([\d.]+)(.*)')
```

Comment:
Compiling a regex is used to create a regex that will be used repeatedly later on.  By compiling it here you are not only creating the new compiled once per method, but once per row iterated through.  This is less efficient than just calling `re` directly on each pass.  Ideally the re.compiled regex would be at the module or class level.